Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -706,6 +724,31 @@ def get_current_speed(self, dir):
         with self.dllock:
             return self.curspeeds[dir]
 
+    def save_resume_data(self):
+        """
+        method to save resume data. It will return either :
+        stored resume data, or deferred if it's not available
+        """
+        ret_val = None
+
+        # prepare deferred entry
+        defer_resume = Deferred()
+        defer_resume.addErrback(lambda _: self._logger.info("Defered resume_data killed before launch"))
```

Comment:
This is the equivalent of a catch all try block. Anything that goes wrong, will be registered as the resume data deferred to be killed before launch.